### ISSUE TITLE
Problem using centreon plugin apps::activedirectory::wsman::plugin

### ISSUE THREAD

**Issue by clairedy** [centreon/centreon-plugins#1973]
Hello,

I'm trying to use centreon plugin apps::activedirectory::wsman::plugin. When i try the command ./centreon_plugins.pl --plugin apps:activedirectory::wsman::plungin --mode=dcdig --hostname=xxx --wsman-auth_method=gssnegociate --wsman-username='xxx' --wsman-password='xxx', i get this error :  UNKNOWN: Can't locate object method "add_option" via package "openwsman::ClientOptions" at /usr/lib64/nagios/plugins/centreon-plugins/centreon/plugins/wsman.pm line 142. 

Do you know what could be missing ?

Best regards,
Claire

**Comment by garnier-quentin** [comment]
could you provide me your openwsman version ? Because the error is quite strange.
wsman plugins are very experimental.

**Comment by clairedy** [comment]
Hi, 

I was using the 2.3.6 version. I upgraded to the 2.6.3 version and now I’m getting this error : UNKNOWN: Could not enumerate instances: Login denied

We are trying to find a plugin that would allow us to monitor Active Directory without an agent.

**Comment by garnier-quentin** [comment]
The error is "ok". You try to connect with an active directory user ?

**Comment by clairedy** [comment]
yes, it's a domain account. We are using kerberos and I did a kinit for this account. We added the account to the groupe Remote Management Users. I'm a now getting this error with debug mode :

avril 30 11:59:01 [62683] Endpoint: http://server:5985/wsman
avril 30 11:59:01 [62683] cl->authentication.verify_peer: 1
avril 30 11:59:01 [62683] *****set post buf len = 1268******
avril 30 11:59:01 [62683] GSS-Negotiate authentication is used
avril 30 11:59:01 [62683] curl error code: 0.
avril 30 11:59:01 [62683] cl->response_code: 500.
avril 30 11:59:01 [62683] cl->last_error code: 0.
avril 30 11:59:01 [62683] NULL response
UNKNOWN: Could not enumerate instances: use debug option to have details

**Comment by garnier-quentin** [comment]
Could you provide your command line ? I think you need to set some options. Maybe it can work with gssnegotiate (also https):
```
--wsman-auth-method=gssnegotiate --wsman-scheme=https
```

Get the complete procedure would be nice if it works.

**Comment by clairedy** [comment]
i'm using this command : 
./centreon_plugins.pl --plugin apps:activedirectory::wsman::plungin --mode=dcdiag --hostname=xxx --wsman-auth-method=gssnegociate --wsman-username='xxx' --wsman-password='xxx'. 

with https, i get a SSL connect error

**Comment by garnier-quentin** [comment]
It tries to check the certificate of your server. maybe it's not valid.

**Comment by clairedy** [comment]
that's normal, since we didn't plan on using https, there is no certificate. 

With another plugin (pywmi_wbem), it works but this plugin only check the system, not the active directory. It's in Python and uses gssapi et wsman for the connexion.

**Comment by garnier-quentin** [comment]
And without https, what error do you have ?
```
--wsman-auth-method=gssnegotiate --wsman-scheme=http
```

**Comment by garnier-quentin** [comment]
You can send me an email and we could do a session to debug it
